src/js/player.js

```diff
@@ -2199,6 +2199,91 @@ class Player extends Component {
     };
   }
 
+  /**
+   * Given a parent control, finds the first focusable child control
+   *
+   * @param {HTMLElement} parent - The control in which we'll search
+   * @return {HTMLElement | null} - Returns the first focusable control in the node list or null if not focusable control is found
+   */
+  getFirstFocusableControl_(parent) {
+    // Parent has no children, and the parent itself is not focusable, nothing to return
+    if (!parent || typeof parent.el !== 'function' || (!Dom.isFocusable(parent.el()) && !parent.children().length)) {
+      return null;
+    }
+
+    if (Dom.isFocusable(parent.el())) {
+      return parent.el();
+    }
+
+    const children = parent.children();
+    let focusableEl;
+
+    for (let i = 0; i < children.length; i++) {
+      const child = children[i];
+
+      focusableEl = this.getFirstFocusableControl_(child);
+
+      if (focusableEl) {
+        break;
+      }
+    }
+
+    return focusableEl;
+  }
+
+  /**
+   * Given a parent control, finds the last focusable child control
+   *
+   * @param {HTMLElement} parent - The control in which we'll search
+   * @return {HTMLElement | null} - Returns the last focusable control in the node list or null if not focusable control is found
+   */
+  getLastFocusableControl_(parent) {
+    // Parent has no children, and the parent itself is not focusable, nothing to return
+    if (!parent || typeof parent.el !== 'function' || !Dom.isVisible(parent.el())) {
+      return null;
+    }
+
+    if (Dom.isFocusable(parent.el())) {
+      return parent.el();
+    }
+
+    const children = parent.children();
+    let focusableEl;
+
+    for (let i = children.length; i >= 0; i--) {
+      const child = children[i];
+
+      focusableEl = this.getLastFocusableControl_(child);
+
+      if (focusableEl) {
+        break;
+      }
+    }
+
+    return focusableEl;
+  }
+
+  trapFullscreenTab_(event) {
+    if (!this._firstFocusableControl || !this._lastFocusableControl) {
+      return;
+    }
+
+    const focusedEl = document.activeElement;
+    let elToFocus;
+
+    if (event.shiftKey) {
+      if (focusedEl === this._firstFocusableControl) {
+        elToFocus = this._lastFocusableControl;
+      }
+    } else if (focusedEl === this._lastFocusableControl) {
+      elToFocus = this._firstFocusableControl;
+    }
+    if (elToFocus && typeof elToFocus.focus === 'function') {
+      elToFocus.focus();
+      event.preventDefault();
+    }
+  }
```

Comment:
Looks like there's a lot of overlap here with the `handleKeyDown` method in `modal-dialog`. I wonder if the code can be shared.